spec/models/user_spec.rb

```diff
@@ -0,0 +1,7 @@
+require 'spec_helper'
+
+describe User do
+	it { should validate_presence_of :email}
+	it { should validate_presence_of :full_name}
+	it { should validate_uniqueness_of :email }
```

Comment:
You are validating more than this. You should test for the other cases you specified in your model - this is not aligned right now.